### PR TITLE
[BEAM-7761] Add transform_name_mapping pipeline option for python sdk

### PR DESCRIPTION
**Pull request by y1chi**
Add transform_name_mapping pipeline option for python sdk.
It is useful for updating an existing pipeline on Google Cloud Dataflow: https://cloud.google.com/dataflow/docs/guides/updating-a-pipeline

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by y1chi** [comment]
R: @aaltay 

Hi ahmet would you mind take a look on this PR?

**Comment by aaltay** [comment]
R; @dustin12 could you take a look at this?

**Comment by y1chi** [comment]
> LGTM.
> 
> Have you tried running a test on Dataflow? Perhaps we can add an integration test for Dataflow.
> 
> Could you also add a JIRA issue to the PR?

tried to add an integration test for dataflow but I guess it's only gonna work after sdk has the option.

File: sdks/python/apache_beam/options/pipeline_options_validator.py [hunk 1]
```diff
@@ -174,6 +179,16 @@ def validate_cloud_options(self, view):
       if not view.job_name:
         errors.extend(self._validate_error(
             'Existing job name must be provided when updating a pipeline.'))
+    if view.transform_name_mapping:
```
Comment (dustin12) [inline sdks/python/apache_beam/options/pipeline_options_validator.py]: Can we also confirm in here that the job is streaming and update.

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 2]
```diff
@@ -476,7 +477,16 @@ def _add_argparse_args(cls, parser):
                         action='store_true',
                         help='Update an existing streaming Cloud Dataflow job. '
                         'Experimental. '
-                        'See https://cloud.google.com/dataflow/pipelines/'
+                        'See https://cloud.google.com/dataflow/docs/guides/'
+                        'updating-a-pipeline')
+    parser.add_argument('--transform_name_mapping',
```
Comment (dustin12) [inline sdks/python/apache_beam/options/pipeline_options.py]: As far as I can tell python does not have a place for streaming specific options.  Is it worth making it and moving this there?

Comment (y1chi) [reply sdks/python/apache_beam/options/pipeline_options.py]: It's probably worth adding generic streaming options, I think this particular option is more dataflow oriented.

Comment (y1chi) [reply sdks/python/apache_beam/options/pipeline_options.py]: This option is probably more update related, which is currently only support with dataflow runner I guess?

File: sdks/python/apache_beam/options/pipeline_options_validator.py (hunk 1 above)
Comment (y1chi) [reply sdks/python/apache_beam/options/pipeline_options_validator.py]: added streaming option validation below.

**Comment by y1chi** [comment]
> I believe our Java streaming numbers integration test and streaming drain test test this option at least somewhat. Does python have the equivalent tests yet?

I'm adding similar update integration tests for python, but did not find any tests related to transform name mapping.

**Comment by y1chi** [comment]
Run Portable_Python PreCommit

**Comment by y1chi** [comment]
Run Python_PVR_Flink PreCommit

File: sdks/python/apache_beam/options/pipeline_options.py (hunk 2 above)
Comment (aaltay) [reply sdks/python/apache_beam/options/pipeline_options.py]: That is correct. Update is only supported by Dataflow.

**Pull request merged**